backend/index.js

```diff
@@ -16,34 +16,39 @@ const finder = new PF.AStarFinder({
 
 setInterval(() => {
   players = players.map((player) => {
-    const p = player;
-
-    if (p.isMoving) {
-      const [x, y] = finder.findPath(
-        p.x,
-        p.y,
-        p.destX,
-        p.destY,
-        grid.clone()
-      )[1];
-
-      if (players.find((pl) => pl.x === x && pl.y === y)) {
-        p.isMoving = false;
-        p.destX = null;
-        p.destY = null;
-      } else {
-        p.x = x;
-        p.y = y;
-
-        if (p.x === p.destX && p.y === p.destY) {
-          p.isMoving = false;
-          p.destX = null;
-          p.destY = null;
+    const playerNew = player;
+
+    if (playerNew.isMoving) {
+      const tempGrid = grid.clone();
```

Comment:
I think it should be enough to copy grid once and then update previous and current position movability instead.